cli/azd/pkg/pipeline/azdo_provider.go

```diff
@@ -223,27 +222,27 @@ func (p *AzdoScmProvider) getRepoDetails() *AzdoRepositoryDetails {
 }
 
 // helper function to return an azuredevops.Connection for use with AzDo Go SDK
-func (p *AzdoScmProvider) getAzdoConnection(ctx context.Context) (*azuredevops.Connection, error) {
-	if p.azdoConnection != nil {
+func (p *AzdoScmProvider) getAzdoConnection(ctx context.Context) (azdo.Connection, error) {
```

Comment:
Same point as above.  It feels more appropriate to keep this as a pointer type in the return.